### PR TITLE
Build ec-cli images for multiple architectures

### PR DESCRIPTION
**Pull request by zregvart**
This builds the ec-cli image for arm64 and ppc64le in addition to amd64.

Additional changes to the image are that ubi-minimal is used, reducing the image by 1/3 (~100MB) and the ENTRYPOINT is specified.

To build on foreign architectures qemu-static must be installed, e.g. on Fedora `qemu-user-static-aarch64` and `qemu-user-static-ppc`.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,9 +14,13 @@
...
+ARG TARGETOS
+ARG TARGETARCH
 
-COPY dist/ec_linux_amd64 /usr/bin/ec
-RUN chmod +x /usr/bin/ec && ec version
+RUN microdnf -y install git-core && microdnf clean all
+
+COPY "dist/ec_"$TARGETOS"_"$TARGETARCH /usr/bin/ec
+
+ENTRYPOINT ["/usr/bin/ec"]
```
Comment (joejstuart) [inline Dockerfile:26]: Will the task need to be updated if we add this?

**Comment by simonbaird** [comment]
Some heavy makefile magic here, perhaps throw in some comments to help the muggles follow along. :smile_cat:

**Comment by zregvart** [comment]
Note that we do not build/push the multi-arch images in the release workflow here, that would be a natural followup on this PR.

**Comment by zregvart** [comment]
This is now ready for review

File: hack/setup-environment.sh [hunk 2]
```diff
@@ -108,7 +108,7 @@ kubectl -n image-registry wait deployment -l "app.kubernetes.io/name=registry" -
 # Build and push the images to the local image registry
 echo -e '✨ \033[1mBuilding images\033[0m'
 make --no-print-directory -C "${ROOT}/.." push-image IMAGE_REPO=localhost:5000/ec PODMAN_OPTS=--tls-verify=false
-make --no-print-directory -C "${ROOT}/.." task-bundle "TASK_REPO=localhost:5000/ec-task-bundle" TASK=<(yq e ".spec.steps[].image? = \"127.0.0.1:5000/ec\"" "${ROOT}"/../task/*/verify-enterprise-contract.yaml)
+make --no-print-directory -C "${ROOT}/.." task-bundle "TASK_REPO=localhost:5000/ec-task-bundle" TASK=<(yq e ".spec.steps[].image? = \"127.0.0.1:5000/ec:latest-$(go env GOOS)-$(go env GOARCH)\"" "${ROOT}"/../task/*/verify-enterprise-contract.yaml)
```
[the remark this answers is not in the document]
Comment (zregvart) [reply hack/setup-environment.sh:111]: `push-image` target used above only builds for a single OS/ARCH...

**Pull request merged**